### PR TITLE
HIVE-27457: Update Kudu table storage handlers in backend db

### PR DESCRIPTION
**Pull request by zchovan**
Before https://issues.apache.org/jira/browse/KUDU-3401 the following table parameters were not filled out when creating a Kudu backed Hive table:

SerDe Library
InputFormat
OutputFormat

This change is adding the necessary update statements to the db schema ddls so that existing Kudu tables can be used after a schema update.

### What changes were proposed in this pull request?
Backend database schema upgrade queries so that Kudu tables are properly handled and can be accessed from Hive.
I've included changes to all supported backend db engines.

### Why are the changes needed?
Before KUDU-3401 was merged, the Kudu client did not fill all necessary fields in the SDS and SERDES tables. This led to Hive being unable to query Kudu backed tables. This change intends to fix the existing Kudu tables, so that they could be used from Hive.

### Does this PR introduce _any_ user-facing change?
No

### Is the change a dependency upgrade?
No

### How was this patch tested?
I tested the changes manually.

### CODE REVIEW

**Comment by akshat0395** [comment]
@zchovan Thanks for the patch, Can you please share how this was tested? 
I feel this change will impact the user, As per my understanding there is no automated job that run upgrade scripts, users will have to manually source this scripts or this script has to be part of one of the version upgrade script(upgrade-x.y.x-to-a.b.c.mysql.sql)
Can we document the steps users need to perform for this change? 
Please correct me if my understanding of this is off

**Comment by zchovan** [comment]
@akshat0395 I manually tested by running the scripts, and running DESCRIBE FORMATTED on a test Kudu table before/after applying, and querying the table itself.

I'll be uploading an additional commit, to include these in the upgrade script. Other than running the upgrade, the user doesn't have to change anything.

**Comment by zchovan** [comment]
@akshat0395 I just checked the upgrade scripts, and there is no upgrade script that includes the changes from HIVE-16399 upwards, e.g. for MSSQL, out of the 36 upgrade scripts, only 25, for MySQL out of 50, only 40 is included. 
Also I found that there are upgrade scripts in the standalone-metastore module as well, but no single upgrade script files there. Should I just append these changes to the end of those scripts as well? Or should these be only appened and removed from the metastore module's scripts?

**Comment by akshat0395** [comment]
Yes, I feel this should be added to the upgrade script. 
@veghlaci05 Can you please check this and guide what would be the right path for applying this change from a liquibase respective as well.

**Review by veghlaci05 — CHANGES_REQUESTED** [review]
The script files in `metastore/scripts/upgrade` are completely outdated except for hive. HSM schema changes must be added in `standalone-metastore/metastore-server/src/main/sql`

You need to modify two files for each db (derby example below): 
- hive-schema-4.0.0-beta-1.derby.sql : This can create the schema from scratch
- upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.derby.sql: Upgrade script from the previous version

As for liquibase: At this point there is no need to take care of that, I will handle the conflicts if that one gets approval.

**Comment by zchovan** [comment]
@veghlaci05 thanks for the help! I've moved the update queries to the standalone-metastore, haven't touched the schema files, as there are no changes to the schema, only the db content.

**Review by veghlaci05 — APPROVED** [review]
LGTM, pending tests

**Review by akshat0395 — APPROVED** [review]

**Comment by akshat0395** [comment]
@zchovan There are multiple test failures, can you please check 
[Tests](http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-4445/3/tests)

File: standalone-metastore/metastore-server/src/main/sql/derby/upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.derby.sql [hunk 1]
```diff
@@ -29,5 +29,21 @@ CREATE INDEX "APP"."PCS_STATS_IDX" ON "APP"."PART_COL_STATS" ("DB_NAME","TABLE_N
 -- HIVE-27186
 ALTER TABLE "APP"."METASTORE_DB_PROPERTIES" ADD PROPERTYCONTENT BLOB;
 
+-- HIVE-27457
+UPDATE SDS
```
Comment (akshat0395) [inline standalone-metastore/metastore-server/src/main/sql/derby/upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.derby.sql:33]: some q test failed, I have attached the link, can you please take a look at those.

**Review by akshat0395 — CHANGES_REQUESTED** [review]

**Comment by akshat0395** [comment]
@zchovan 1 test failing on the Postgres side, can you please check. Thats the last one pending
[TestPostgres]( http://ci.hive.apache.org/blue/organizations/jenkins/hive-precommit/detail/PR-4445/4/tests)

**Comment by akshat0395** [comment]
All test passed, LGTM +1

**Review by akshat0395 — APPROVED** [review]

**Pull request merged**